schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/spi/ActivityEndpoint.java

```diff
@@ -98,8 +98,10 @@ public ActivityType updateActivityType(User caller, @Named("id") Key id, Activit
         try {
             final ActivityType dbActivityType = activityTypeDao.get(activityType.getId());
             final Key organizationId = dbActivityType.getOrganizationRef().getKey();
-            if (activityTypeDao.exists(activityType.getName(), organizationId)) {
-                throw new BadRequestException("ActivityType.name=" + activityType.getName() + ", Organization.id=" + organizationId);
+            if (!StringUtils.equalsIgnoreCase(dbActivityType.getName(), activityType.getName())) {
+                if (activityTypeDao.exists(activityType.getName(), organizationId)) {
+                    throw new BadRequestException("ActivityType.name=" + activityType.getName() + ", Organization.id=" + organizationId);
+                }
             }
             return ActivityServiceFactory.getActivityService().updateActivityType(activityType);
```

Comment:
Whenever you replace the update with the dao, you should wrap this in a transaction (with transaction operator).  Else you could end up with two activity types with the same name.